packages/app-project/src/screens/ClassifyPage/components/YourStats/YourStats.spec.js

```diff
@@ -1,15 +1,41 @@
-import { shallow } from 'enzyme'
+import { render, screen } from '@testing-library/react'
+import { composeStory } from '@storybook/react'
+import Meta, { YourStats } from './YourStats.stories.js'
+import { YourStatsStoreMock } from './YourStats.mock'
 
-import YourStats from './YourStats'
+describe('Component > YourStats', function () {
+  beforeEach(function () {
+		const YourStatsStory = composeStory(YourStats, Meta)
+		render(<YourStatsStory />)
+	})
 
-let wrapper
+	it('should have a heading', function () {
+    expect(screen.findByText(`Your ${YourStatsStoreMock.project.display_name} Statistics`)).to.be.ok()
+  })
```

Comment:
**Note 1:** I don't think `findByText()` can be used this way.

findByText() returns a _Promise,_ not the _element_ you're looking for. [(docs)](https://testing-library.com/docs/queries/about/#types-of-queries). So if you replaced the text with "NONSENSE TEXT", then this happens...

```
expect(screen.findByText('NONSENSE TEXT')).to.be.ok()
=> expect(Promise).to.be.ok()
=> a Promise is an object, therefore it's OK
=> test passes
``` 

So this test passes no matter what text you're trying to find. 😱 

**Note 1a:** Also! The current RTL setup _doesn't actually render the translations,..

So if you run `screen.debug()`, you'll see that the heading isn't rendered as `Your Stats Mock Project Statistics` (as you'd see on Storybook), but instead as `Classify.YourStats.title` (i.e. it renders the translation key)

**Suggested Solution**

Option 1: if you need to use findBy...(), make sure to use await.

For example,

```
it('should have a heading', async function () {
  const heading = await screen.findByText('Classify.YourStats.title')
  expect(heading).to.be.ok()
})
```

⭐  Option 2: the simpler way is to use getBy...() _(recommended!)_

getByText() will either find the component you're looking for, or throw an error if _either 0 or > 1 elements_ are found. [(docs)](https://testing-library.com/docs/queries/about/#types-of-queries) For example,

```
it('should have a heading', function () {
  expect(screen.getByText(`Classify.YourStats.title`)).to.be.ok()
})
```

Easy!

Reply:
Addendum: when we're looking for exactly one element, then in most cases getByText() works pretty much the same as queryByText(). So using either one of them should be fine, AFAIK.

findByText() however, since it returns a Promise, seems more situational - if you look at the other unit tests in `app-project`, it's findBy...() calls are _only_ used in conjunction with the more complex event or user input-driven tests. (e.g. DropdownNav, LocaleSwitcher, etc)

Reply:
Thanks for the insight on how Promises are handled within RTL.. I assumed that they were resolved before evaluated :/

Fixed using the translation keys.